### ISSUE TITLE
Import Inception like models in DL4J

### ISSUE THREAD

**Issue by suyashps** [deeplearning4j/deeplearning4j#2362]
As discussed in https://github.com/deeplearning4j/deeplearning4j/issues/2035. How difficult it would be to support TensorFlow?

**Comment by turambar** [comment]
@suyashps Based on a quick glance, the only thing missing from our [Keras model import](https://github.com/deeplearning4j/deeplearning4j/tree/master/deeplearning4j-modelimport) to fully support importing [Inception](https://github.com/fchollet/keras/blob/master/keras/applications/inception_v3.py) models is batch normalization. That should be an easy add.

**Comment by turambar** [comment]
Importing non-Keras TensorFlow models is another matter. To import a _model architecture_, you need to be able to save your TF model architecture into a standardized configuration file. That is what [Keras provides](https://keras.io/getting-started/faq/#how-can-i-save-a-keras-model), which is why we recommend using it to construct models that you wish to import into DL4J. You can see in our [Keras model configuration import](https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/ModelConfiguration.java) module that we depend crucially on Keras having a well-defined configuration schema written into a portable file format (JSON)

Once you have an identically configured Inception architecture in DL4J (whether imported or manually configured), importing weights is pretty straightforward provided you know how to match layer and parameter names and you've saved your weights into a portable format, such as HDF5 (again, as Keras does). In fact, you if you follow the Keras' HDF5 schema for saving weights, you can directly use our Keras weight import functionality.

**Comment by turambar** [comment]
@suyashps We're making progress on this. The [model_loading](https://github.com/deeplearning4j/deeplearning4j/tree/model_loading) branch now supports the Functional API (graph structured Keras models with model class name "Model"). We're working on support for the BatchNormalization layer, which should be ready in a day or two.

**Comment by suyashps** [comment]
@turambar Thats a good news! I will check it out. Does it have the BatchNormalization support now?

**Comment by turambar** [comment]
@suyashps use with the `master` branch now -- @eraly made some important commits. A warning, though: Keras models with a Convolutional layer that used the TensorFlow backend are currently broken. Fix coming soon.
